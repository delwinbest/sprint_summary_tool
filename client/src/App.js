import React from "react";
import { useSelector } from "react-redux";
import { Route, Switch, Redirect, BrowserRouter } from "react-router-dom";

import AuthLayout from "layouts/Auth.js";
// import RtlLayout from "layouts/RTL.js";
import AdminLayout from "layouts/Admin.js";

import "assets/scss/material-dashboard-pro-react.scss?v=1.10.0";

function App() {
  const { id: userId } = useSelector((state) => state.user);
  return (
    <BrowserRouter>
      <Switch>
        {/* <Route path="/rtl" component={RtlLayout} /> */}
        <Route path="/auth" component={AuthLayout} />
        {userId && <Route path="/admin" component={AdminLayout} />}
        {userId ? <Redirect to="/admin/dashboard" /> : <Redirect to="/auth" />}
        <Redirect to="/auth" />
      </Switch>
    </BrowserRouter>
  );
}

export default App;
