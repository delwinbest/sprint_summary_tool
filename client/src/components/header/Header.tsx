import { Navbar, Container } from 'react-bootstrap';
import { Bricks } from 'react-bootstrap-icons';

const Header = () => {
  return (
    <Navbar bg="dark" variant="dark">
      <Container>
        <Navbar.Brand href="#home">
          <Bricks color="beige" /> Sprint Planner
        </Navbar.Brand>
      </Container>
    </Navbar>
  );
};

export default Header;
