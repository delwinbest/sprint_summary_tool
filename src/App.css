.table td {
  text-align: center;
  align-content: center !important;
  vertical-align: middle !important;
}
