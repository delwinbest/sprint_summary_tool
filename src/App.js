import React, { useState, useRef } from "react";
import {
  Button,
  Row,
  Card,
  Badge,
  Container,
  Modal,
  InputGroup,
  Form,
  ListGroup,
} from "react-bootstrap";
import { useSelector, useDispatch } from "react-redux";
import * as actionTypes from "./store/actions/actionTypes";

function App() {
  const dispatch = useDispatch();
  const team = useSelector((state) => state.team);

  const [showModal, setShowModal] = useState(false);
  const newMemberRef = useRef();

  const handleModalClose = () => setShowModal(false);
  const handleModalShow = () => setShowModal(true);

  const addMemberHandler = (username) => {
    dispatch({ type: actionTypes.TEAM_ADD_MEMBER, username: username });
  };

  const removeMemberHandler = (username) => {
    dispatch({ type: actionTypes.TEAM_REMOVE_MEMBER, username: username });
  };

  const teamList = team.teamMembers.map((employee) => {
    return (
      <ListGroup.Item key={employee}>
        <Badge variant="danger" onClick={() => removeMemberHandler(employee)}>
          -
        </Badge>
        <span style={{ "padding-left": "1rem" }}>{employee}</span>
      </ListGroup.Item>
    );
  });

  return (
    <Container>
      <Row className="justify-content-md-center">
        <Card style={{ width: "18rem" }}>
          <Card.Header>Team Members</Card.Header>
          <ListGroup variant="flush">{teamList} </ListGroup>
          <Button variant="outline-success" onClick={handleModalShow}>
            Add New Member
          </Button>
        </Card>
      </Row>

      <Modal show={showModal} onHide={handleModalClose}>
        <Modal.Header closeButton>
          <Modal.Title>Add Team Member</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <InputGroup className="mb-3">
            <Form.Control
              placeholder="username"
              aria-label="username"
              aria-describedby="basic-addon2"
              ref={newMemberRef}
            />
            <InputGroup.Append>
              <InputGroup.Text id="basic-addon2">@amazon.com</InputGroup.Text>
            </InputGroup.Append>
          </InputGroup>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={handleModalClose}>
            Close
          </Button>
          <Button
            variant="primary"
            onClick={() => {
              addMemberHandler(newMemberRef.current.value);
              handleModalClose();
            }}
          >
            Save Changes
          </Button>
        </Modal.Footer>
      </Modal>
    </Container>
  );
}

export default App;
