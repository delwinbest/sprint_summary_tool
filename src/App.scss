.table td {
  text-align: center;
  align-content: center !important;
  vertical-align: middle !important;
}

thead {
  th {
    font-size: 0.9rem !important;
  }
}

td {
  input {
    text-align: center !important;
    color: white !important;
    background-color: transparent !important;
    border-color: transparent !important;
    font-size: 0.9rem !important;

    &:hover {
      background-color: white !important;
      color: var(--dark) !important;
    }
    &::placeholder {
      color: white !important;
      opacity: 1 !important;
    }
  }
}

.dropdown-menu {
  .dropdown-item {
    &.active {
      background-color: rgb(23, 162, 184) !important;
    }
  }
}
