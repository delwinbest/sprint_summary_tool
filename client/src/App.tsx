import React from 'react';
import { Container } from 'react-bootstrap';
import './App.css';
import Header from './components/header/Header';

function App() {
  return (
    <div className="App">
      <Container>
        <Header />
      </Container>
    </div>
  );
}

export default App;
