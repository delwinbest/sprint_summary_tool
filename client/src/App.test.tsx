import React from 'react';
import { shallow, mount } from 'enzyme';
import App from './App';
import toJson from 'enzyme-to-json';
import Header from './components/header/Header';

it('renders without crashing', () => {
  shallow(<App />);
});

// it('renders correctly with no error message', () => {
//   const wrapper = mount();
//   expect(wrapper.state('error')).toEqual(null);
// });

// describe('', () => {
//   it('accepts user account props', () => {
//     const wrapper = mount(<Account user={user} />);
//     expect(wrapper.props().user).toEqual(user);
//   });
//   it('contains users account email', () => {
//     const wrapper = mount(<Account user={user} />);
//     const value = wrapper.find('p').text();
//     expect(value).toEqual('david@gmail.com');
//   });
// });

it('renders correctly', () => {
  const tree = shallow(<App />);
  expect(toJson(tree)).toMatchSnapshot();
});

it('renders Header component', () => {
  const wrapper = shallow(<App />);
  const header = <Header />;
  expect(wrapper.contains(header)).toEqual(true);
});
