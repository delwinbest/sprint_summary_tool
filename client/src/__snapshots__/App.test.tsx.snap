// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <Container
    fluid={false}
  >
    <Header />
  </Container>
</div>
`;
